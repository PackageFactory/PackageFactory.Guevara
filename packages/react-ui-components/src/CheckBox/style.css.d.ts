export interface IStyleCss {
  'checkbox': string;
  'checkbox__input': string;
  'checkbox__inputMirror': string;
  'label': string;
  'checkbox__inputMirror--active': string;
  'checkbox__disabled': string;
}

export const locals: IStyleCss;
