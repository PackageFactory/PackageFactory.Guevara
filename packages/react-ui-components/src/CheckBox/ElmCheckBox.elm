port module ElmCheckBox exposing (main)

import Html exposing (..)
import Html.Attributes exposing (..)
import Html.Events exposing (..)


main : Program Flags Model Msg
main =
    Html.programWithFlags
        { init = init
        , view = view
        , update = update
        , subscriptions = subscriptions
        }



-- MODEL


type alias Flags =
    { isChecked : Bool
    , isDisabled : Maybe Bool
    , highlight : Maybe Bool
    , className : Maybe String
    }


type alias Model =
    { isChecked : Bool
    , isDisabled : Bool
    , highlight : Bool
    , className : String
    }


init : Flags -> ( Model, Cmd Msg )
init flags =
    let
        isChecked =
            flags.isChecked

        isDisabled =
            case flags.isDisabled of
                Just disabled ->
                    True

                Nothing ->
                    False

        highlight =
            case flags.highlight of
                Just highlight ->
                    True

                Nothing ->
                    False

        className =
            case flags.className of
                Just className ->
                    className

                Nothing ->
                    ""
    in
    ( { isChecked = isChecked
      , isDisabled = isDisabled
      , highlight = highlight
      , className = className
      }
    , Cmd.none
    )



-- UPDATE


type Msg
    = OnClick


update : Msg -> Model -> ( Model, Cmd Msg )
update msg model =
    case msg of
        OnClick ->
            ( { model | isChecked = not model.isChecked }, Cmd.batch [ handleClick model.isChecked ] )


port handleClick : Bool -> Cmd msg



-- SUBSCRIPTIONS


subscriptions : Model -> Sub Msg
subscriptions model =
    Sub.none



-- VIEW


myStyle : Attribute msg
myStyle =
    style
        [ ( "position", "absolute" )
        , ( "top", "50%" )
        , ( "left", "0" )
        , ( "z-index", "3" )
        , ( "transform", "translateY(-50%)" )
        , ( "width", "20px" )
        , ( "height", "20px" )
        , ( "opacity", "0" )
        , ( "cursor", "pointer" )
        ]


view : Model -> Html Msg
view model =
    input [ myStyle, type_ "checkbox", checked model.isChecked, disabled model.isDisabled, onClick OnClick ] []
