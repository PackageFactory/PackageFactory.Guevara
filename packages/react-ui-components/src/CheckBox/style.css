.checkbox {
    composes: reset from './../reset.css';
    position: relative;
    display: inline-block;
    margin-right: 8px;
    vertical-align: middle;
}
.checkbox__input {
    composes: reset from './../reset.css';
    position: absolute;
    top: 50%;
    left: 0;
    z-index: var(--zIndex-CheckboxInput-Context);
    transform: translateY(-50%);
    width: 20px;
    height: 20px;
    opacity: 0;
    cursor: pointer;
}
.checkbox__inputMirror {
    composes: reset from './../reset.css';
    display: inline-block;
    width: 20px;
    height: 20px;
    border: 1px solid var(--colors-ContrastBrighter);
    vertical-align: text-bottom;
    background: var(--colors-ContrastDark);
    font-size: 20px;
    font-weight: bold;
    color: #5b5b5b;

    .label:hover & {
        border-color: var(--colors-PrimaryBlue);
    }

    &:before {
        content: '\2713';
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translateX(-50%) translateY(-50%);
    }
}
.checkbox__inputMirror--active {
    background: var(--colors-PrimaryBlue);
    border-color: var(--colors-PrimaryBlue);
    color: #fff;

    &:before {
        content: '\2713';
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translateX(-50%) translateY(-50%);
    }
}
.checkbox__disabled {
    opacity: .65;
    cursor: not-allowed;

    .checkbox__input {
        cursor: not-allowed;
    }
}
