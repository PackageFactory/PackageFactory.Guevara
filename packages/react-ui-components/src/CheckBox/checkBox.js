import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import mergeClassNames from 'classnames';
import Elm from 'react-elm-components';
import ElmCheckBox from './ElmCheckBox.elm';

class CheckBox extends PureComponent {
    static propTypes = {
        /**
         * This prop controls the visual active state of the CheckBox.
         */
        isChecked: PropTypes.bool.isRequired,

        /**
         * This prop controls if the CheckBox is disabled or not.
         */
        isDisabled: PropTypes.bool,

        /**
         * An optional `className` to attach to the wrapper.
         */
        className: PropTypes.string,

        /**
         * An optional change handler which gets called once the user clicks on the CheckBox.
         */
        onChange: PropTypes.func,

        /**
         * Highlight input
         */
        highlight: PropTypes.bool,

        /**
         * An optional css theme to be injected.
         */
        theme: PropTypes.shape({
            'checkbox': PropTypes.string,
            'checkbox__input': PropTypes.string,
            'checkbox__inputMirror': PropTypes.string,
            'checkbox__inputMirror--active': PropTypes.string,
            'checkbox__inputMirror--highlight': PropTypes.string
        }).isRequired
    };

    handleChange = isChecked => {
        const {onChange} = this.props;

        if (onChange) {
            onChange(!isChecked);
        }
    }

    render() {
        const {
            isChecked,
            isDisabled,
            className,
            theme,
            highlight
        } = this.props;
        const finalClassName = mergeClassNames({
            [className]: className && className.length,
            [theme.checkbox]: true,
            [theme.checkbox__disabled]: isDisabled
        });
        const mirrorClassNames = mergeClassNames({
            [theme.checkbox__inputMirror]: true,
            [theme['checkbox__inputMirror--active']]: isChecked,
            [theme['checkbox__inputMirror--highlight-checked']]: highlight && isChecked,
            [theme['checkbox__inputMirror--highlight-unchecked']]: highlight && !isChecked
        });

        const flags = {
            isChecked,
            isDisabled: isDisabled || null,
            highlight: highlight || null,
            className: className ? className : null
        };

        const setupPorts = (ports => {
            if (ports) {
                ports.handleClick.subscribe(isChecked => {
                    this.handleChange(isChecked);
                });
            }
        });

        return (
            <div className={finalClassName}>
                <Elm src={ElmCheckBox} flags={flags} ports={setupPorts} />
                <div className={mirrorClassNames}/>
            </div>
        );
    }
}

export default CheckBox;
