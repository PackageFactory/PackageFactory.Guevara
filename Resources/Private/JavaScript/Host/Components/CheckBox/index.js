import React, {PropTypes} from 'react';
import mergeClassNames from 'classnames';
import uuid from 'uuid';
import Label from 'Host/Components/Label/';
import style from './style.css';

const onChangeHandler = (cb, isChecked) => {
    if (cb) {
        cb(isChecked);
    }
};

const CheckBox = props => {
    const {
        isChecked,
        label,
        className,
        onChange
    } = props;
    const classNames = mergeClassNames({
        [className]: className && className.length,
        [style.checkbox]: true
    });
    const mirrorClassNames = mergeClassNames({
        [style.checkbox__inputMirror]: true,
        [style['checkbox__inputMirror--active']]: isChecked
    });
    const id = uuid.v1();

    return (
        <Label
            className={style.label}
            htmlFor={id}
            label={label}
            isChildrenInlined={true}
            labelPosition="after"
            >
                <div className={classNames}>
                <input
                    id={id}
                    className={style.checkbox__input}
                    type="checkbox" checked={isChecked}
                    onChange={() => onChange(onChangeHandler, !isChecked)}
                    />
                <div className={mirrorClassNames}></div>
            </div>
        </Label>
    );
};
CheckBox.propTypes = {
    // State related propTypes.
    isChecked: PropTypes.bool.isRequired,

    // Style related propTypes.
    className: PropTypes.string,

    // Content related propTypes.
    label: PropTypes.string.isRequired,

    // Interaction related propTypes.
    onChange: PropTypes.func
};

export default CheckBox;
