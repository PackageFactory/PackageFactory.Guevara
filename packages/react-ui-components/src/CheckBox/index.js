import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from './../identifiers';
import style from './style.css';
import CheckBox from './checkBox.js';

export default themr(identifiers.checkBox, style)(CheckBox);
