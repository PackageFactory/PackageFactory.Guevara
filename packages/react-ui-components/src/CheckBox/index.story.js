import React from 'react';
import {storiesOf, action} from '@storybook/react';
import {withKnobs, boolean} from '@storybook/addon-knobs';
import {StoryWrapper} from './../_lib/storyUtils.js';
import CheckBox from './index.js';

storiesOf('CheckBox', module)
    .addDecorator(withKnobs)
    .addWithInfo(
        'default',
        'Just a checkbox. Can be checked or not.',
        () => (
            <StoryWrapper>
                <CheckBox onChange={action('onChange')} isChecked={boolean('Checked', true)}/>
            </StoryWrapper>
        ),
        {inline: true}
    );
