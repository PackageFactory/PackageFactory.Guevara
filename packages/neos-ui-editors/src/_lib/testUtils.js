import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
