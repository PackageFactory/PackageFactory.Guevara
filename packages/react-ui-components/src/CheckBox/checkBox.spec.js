import sinon from 'sinon';
import {createShallowRenderer} from './../_lib/testUtils.js';
import CheckBox from './checkBox.js';

const defaultProps = {
    isChecked: false,
    theme: {}
};
const shallow = createShallowRenderer(CheckBox, defaultProps);

test('should render a "input" node with the role="button" attribute.', () => {
    const input = shallow().find('[type="checkbox"]');

    expect(input.length).toBe(1);
});
test('should render the "className" prop if passed.', () => {
    const input = shallow({
        className: 'barClassName'
    });

    expect(input.hasClass('barClassName')).toBeTruthy();
});
test('should throw no errors if no "onChange" prop was passed when clicking on the hidden checkbox.', () => {
    const cb = shallow();
    const fn = () => cb.find('[type="checkbox"]').simulate('change');

    expect(fn).not.toThrow();
});
test('should call the passed "onChange" prop when clicking on the hidden checkbox.', () => {
    const onChange = sinon.spy();
    const cb = shallow({onChange});

    cb.find('[type="checkbox"]').simulate('change');

    expect(onChange.callCount).toBe(1);
});
test('should set truthy aria and checked attribute when passing a truthy "isChecked" prop.', () => {
    const markup = shallow({isChecked: true}).find('[type="checkbox"]').html();

    expect(markup.includes('checked="true"')).toBeTruthy();
    expect(markup.includes('aria-checked="true"')).toBeTruthy();
});
test('should set falsy aria and checked attribute when passing a falsy "isChecked" prop.', () => {
    const markup = shallow({isChecked: false}).find('[type="checkbox"]').html();

    expect(markup.includes('checked="false"')).toBeTruthy();
    expect(markup.includes('aria-checked="false"')).toBeTruthy();
});
