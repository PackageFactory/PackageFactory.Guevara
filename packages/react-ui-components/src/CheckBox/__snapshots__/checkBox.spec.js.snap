// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox/> should render correctly. 1`] = `
<div
  className=""
>
  <input
    aria-checked={false}
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className=""
  />
</div>
`;
